addons/survey/tests/test_survey_performance.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import time
+
+from logging import getLogger
+
+from odoo.addons.survey.tests import common
+from odoo.addons.website.tests.test_performance import UtilPerf
+from odoo.tests import tagged
+from odoo.tests.common import warmup
+
+_logger = getLogger(__name__)
+
+
+@tagged('post_install', '-at_install')
+class SurveyPerformance(common.TestSurveyResultsCommon, UtilPerf):
+
+    @warmup
+    def test_survey_results_with_one_filter(self):
+        url = f'/survey/results/{self.survey.id}?filters=A,0,{self.cat_id}'
+        self.authenticate('survey_manager', 'survey_manager')
+        with self.assertQueryCount(default=32): # com 32
+            self.url_open(url)
+
+    @warmup
+    def test_survey_results_with_multiple_filters_question_answer_model(self):
+        """ Check that no matter the number of filters, if their answers
+        data are stored in the same model (here survey.question.answer)
+        the query count stay the same as having a single filter.
+        """
+        url = f'/survey/results/{self.survey.id}?filters=A,0,{self.gras_id}|A,0,{self.cat_id}'
+        self.authenticate('survey_manager', 'survey_manager')
+        with self.assertQueryCount(default=32): # com 32
+            self.url_open(url)
+
+    @warmup
+    def test_survey_results_with_multiple_filters_mixed_model(self):
+        """ Check that, in comparison with having filters from the same model,
+        having filters from different models needs only a few more queries.
+        """
+        url = f'/survey/results/{self.survey.id}?filters=A,0,{self.gras_id}|L,0,{self.answer_pauline.id}'
+        self.authenticate('survey_manager', 'survey_manager')
+        t0 = time.time()
+        with self.assertQueryCount(default=34): # com 34
+            self.url_open(url)
+        t1 = time.time()
+        _logger.info('Browsed %s, time %.3f', url, t1 - t0)
```

Comment:
Not sure that is useful as we are most likely not going to look at those numbers.